File: src/persistence/memory.ts
/**
 * In-Memory Storage Adapter
 *
 * Simple key-value storage that lives only for the process lifetime.
 * Useful for testing, serverless functions, or when persistence isn't needed.
 *
 * @module persistence/memory
 */

/**
 * In-memory storage adapter for ApiKeyManager.
 *
 * @example
 * ```ts
 * import { ApiKeyManager } from '@splashcodex/api-key-manager';
 * import { MemoryStorage } from '@splashcodex/api-key-manager/persistence/memory';
 *
 * const manager = new ApiKeyManager(keys, {
 *   storage: new MemoryStorage()
 * });
 * ```
 */
export class MemoryStorage {
    private store: Map<string, string> = new Map();

    getItem(key: string): string | null {
        return this.store.get(key) ?? null;
    }

    setItem(key: string, value: string): void {
        this.store.set(key, value);
    }

    /** Clear all stored state */
    clear(): void {
        this.store.clear();
    }

    /** Get the number of stored entries */
    get size(): number {
        return this.store.size;
    }
}

File: python-sdk/splashcodex_api_manager/persistence/memory.py
from typing import Optional, Dict

class MemoryStorage:
    """
    In-Memory Storage Adapter

    Simple key-value storage that lives only for the process lifetime.
    Useful for testing, serverless functions, or when persistence isn't needed.
    """
    def __init__(self):
        self._store: Dict[str, str] = {}

    def get_item(self, key: str) -> Optional[str]:
        return self._store.get(key)

    def set_item(self, key: str, value: str) -> None:
        self._store[key] = value

    def clear(self) -> None:
        self._store.clear()

    @property
    def size(self) -> int:
        return len(self._store)
